toolchain/semantics/semantics_ir.cpp

```diff
@@ -404,6 +413,7 @@ auto SemanticsIR::StringifyType(SemanticsTypeId type_id,
   if (!in_type_context) {
     auto outer_node = GetNode(outer_node_id);
     if (outer_node.kind() == SemanticsNodeKind::TupleType ||
+        outer_node.kind() == SemanticsNodeKind::ArrayType ||
```

Comment:
I think we don't need this change: `[T; N]` already gives us a value of type `type`, so we don't need an `as type` to convert it.